text/XXXX-externalize_private_keys.md

```diff
@@ -0,0 +1,111 @@
+
+### Recovering from a deleted `.qri` folder
+Once we've externalized the private keys, we can change the behaviour of `qri setup` to first scan the keyring under the `io.qri.keys` namespace for existing keys. If a key is found, we can ask the user if they'd like to create a new account, or recover an old one.
+
+We can bolster this with calls to the default registry for the discovered public keys to add username info to the list. Profile information may not exist on the default registry, so we'll need to default to showing just the raw ProfileID. We can support this process with docs.
+
+Instead of jumping straight to building the UI for switching, we can supply a `--recover` flag that lets a user supply a ProfileID to recover from:
+
+```
+$ qri setup --recover QmProfileID
+```
+
+listing available private keys in a keyring may also form the basis of profile switching in the future.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+### More OS-Specific Work.
+This increases our reliance on an OS-level API that sometimes doesn't exist, and runs the risk of deviating behaviour between platforms. 
+
+Fallback to a `QRI_PRIVATE_KEY` file whenever no keyring exists is necessary to keep things working when our keyring implementation fails, effectivly reverting us back to the setup we're in today. This time we at least have a separate file and a warning.
+
+### Upstream work to get recovery functionality
+The package I've suggested we build on has no high-level method for listing the keys in a namespace. We'e need to be able to do this, and haven't confirmed this will be possible across all platforms.
+
+### No Qri-specific security
+This RFC does not propose adding any key-specific security. If we used symmetric encryption on the private key itself, Qri could require a local password of it's own. In this case we're relying on the user's Operating System security architecture
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+A number of alternatives exist for externalizing secrets. We could store keys in the `.ssh` directory if it exists, and looking for guidance on 
+
+# Prior art
+[prior-art]: #prior-art
+
+* Keybase - We should really do a dive on the approach keybase uses for key management before moving on to any other keypair work. _Especially_ key rotation.
+* Textile wallet
+* Hashicorp Vault
+
+* https://github.com/keybase/go-keychain
+* https://github.com/zalando/go-keyring
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+### Loading order for configuration
+We haven't defined _when_ key loading happens in this process now that key loading is separate from configuration loading in the relation to seting up other services.
+
+### Profile Switching
+This RFC lays the groundwork for switching accounts _within_ qri, but this would have very strange results, as the data stored in `.qri` is mapped to a single user. We also don't have any UI for how this would work. 
```

Comment:
I very much like this idea from the user perspective. Assuming I'm a heavy user and use it for 1 or more orgs and also personally, it gives a great way of separating contexts with users essentially working as a namespace in this scenario. This makes even more sense when we consider enterprise users/private data networks coupled with regular usage of qri on the public network.

Reply:
Totally. To that end I could see us evolving the `.qri` folder to shard into directories of repos, with each directory being a different identity. We could use the current `.qri` contents as a "primary user".